src/Worker/ReschedulingDecorator/Builder.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neighborhoods\KojoWorkerDecoratorComponent\Worker\ReschedulingDecorator;
+
+use LogicException;
+use Neighborhoods\Kojo\Api;
+use Neighborhoods\KojoWorkerDecoratorComponent\Worker\DecoratorInterface;
+use Neighborhoods\KojoWorkerDecoratorComponent\Worker;
+
+class Builder implements BuilderInterface
+{
+    use Api\V1\Worker\Service\AwareTrait;
+    use Api\V1\RDBMS\Connection\Service\AwareTrait;
+    use Factory\AwareTrait;
+    use Worker\AwareTrait;
+
+    private $prefabDoctrineDbalConnectionDecoratorRepository;
+    private /*string*/ $connectionId;
+
+    public function build(): DecoratorInterface
+    {
+        $decorator = $this->getWorkerReschedulingDecoratorFactory()
+            ->create();
+
+        $connection = $this->getPrefabDoctrineDbalConnectionDecoratorRepository()
+            ->getConnection($this->getConnectionId());
+
+        $decorator->setApiV1RDBMSConnectionService($this->getApiV1RDBMSConnectionService());
```

Comment:
**blocker**

This isn't defined by the interface, nor is it used by the decorator

Reply:
[ReschedulingDecoratorInterface](https://github.com/neighborhoods/KojoWorkerDecoratorComponent/blob/rescheduling/src/Worker/ReschedulingDecoratorInterface.php) extends [DecoratorInterface](https://github.com/neighborhoods/KojoWorkerDecoratorComponent/blob/rescheduling/src/Worker/DecoratorInterface.php), which in turn extends [WorkerInterface](https://github.com/neighborhoods/KojoWorkerDecoratorComponent/blob/rescheduling/src/WorkerInterface.php), which has the `setApiV1RDBMSConnectionService` method.

But it's a good point. It's because the two kojo services get provided by kojo when it runs the worker. We don't have them before and cannot inject them during container building. We also cannot assume if the worker/decorator needs any of them, so we inject them into every one to make them accessible.

I feel like this is the same story as https://github.com/neighborhoods/KojoWorkerDecoratorComponent/issues/4 and could be solved using the proposed approach. The difference is that the the logger isn't kojo specific and therefore doesn't belong in this component, but these two services are so it makes sense to define their placeholders here.
I can make a separate branch to demonstrate how I imagine it if that's not clear.